From 04008befad133d97b7f956a21632592aaaede7ca Mon Sep 17 00:00:00 2001
From: CTCaer <ctcaer@gmail.com>
Date: Sat, 11 Jun 2022 17:14:49 +0000
Subject: [PATCH 15/20] codecs: nvv4l2: BSP 34.1.x remarks

NVIDIA changed the enum again in r34 and broke compatibility for no reason.

So NvBufferColorFormat_YUV444 enum value changed and nvbuf_utils needs a new heuristic to support it.
Anything else should work as is.
---
 libavcodec/nvv4l2.c           | 1 +
 libavcodec/nvv4l2_ext_utils.h | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/libavcodec/nvv4l2.c b/libavcodec/nvv4l2.c
index 9c63a88fed..44a825c8c2 100644
--- a/libavcodec/nvv4l2.c
+++ b/libavcodec/nvv4l2.c
@@ -816,6 +816,7 @@ nvv4l2_dbg_plane_supported_formats(nvv4l2_ctx_t *ctx,
     }
 }
 
+/*! TODO: Add 34.1.x support; YUV444 enum value changed. */
 NvBufferPixFmtVersion
 nvv4l2_get_pixfmt_list_version(nvv4l2_ctx_t *ctx)
 {
diff --git a/libavcodec/nvv4l2_ext_utils.h b/libavcodec/nvv4l2_ext_utils.h
index 3f1866ad69..80918497f1 100644
--- a/libavcodec/nvv4l2_ext_utils.h
+++ b/libavcodec/nvv4l2_ext_utils.h
@@ -2145,7 +2145,7 @@ typedef enum
   /** BT.601 colorspace - Y/CbCr 4:2:0 10-bit multi-planar. */
   NvBufferColorFormat_NV12_10LE = 19, /* BSP 32.5.0 and up: 20 */
   /** BT.601 colorspace - YUV444 multi-planar. */
-  NvBufferColorFormat_YUV444    = 35, /* BSP 32.5.0 and up: 36 */
+  NvBufferColorFormat_YUV444    = 35, /* BSP 32.5.0 and up: 36. BSP 34.1.0 and up: 34 */
 } NvBufferColorFormat;
 
 /**
-- 
2.25.1

